test/distributed/test_c10d_nccl.py

```diff
@@ -3532,6 +3537,15 @@ def test_new_group_local_sync_sanity_check(self):
     def test_new_group_local_sync_duplicated_pg(self):
         self._test_new_group_local_sync_duplicate_pg(backend="nccl")
 
+    def _init_pg_subgroups(self, world_size: int):
```

Comment:
nit: its a little bad to expose an 'int' arg for world_size, but then also assume it's gonna be 4 (hardcoding 0,1 and 2,3).

suggest either naming the function about '2x2 subgroups' and then hardcode/assert 4, or, put the logic in here to take world_size and make 2 groups of [:size/2] and [size/2:] ranks.

(even if you make the helper parametrized on world_size, you can still leave the tests themselves hardcoded to 4 for simplicity)

Reply:
good point. changed to ``_init_two_pg2_subgroups`` and assert on ``world_size==4``